src/runtime/WeakRefObject.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2021-present Samsung Electronics Co., Ltd
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
+ *  USA
+ */
+
+#ifndef __EscargotWeakRefObject__
+#define __EscargotWeakRefObject__
+
+#include "runtime/Object.h"
+
+namespace Escargot {
+
+class WeakRefObject : public Object {
+public:
+    explicit WeakRefObject(ExecutionState& state, Object* target);
+    explicit WeakRefObject(ExecutionState& state, Object* proto, Object* target);
+
+    virtual bool isWeakRefObject() const
+    {
+        return true;
+    }
+
+    Object* getTarget()
+    {
+        if (m_target) {
+            return m_target;
+        }
+        return nullptr;
+    }
```

Comment:
How about changing the return type of `getTarget` to `Value` ?
If so, we can simplify `getTarget` call in `builtinWeakRefDeRef`.

Reply:
or You can use Optional<Object*> for clean code

Reply:
Good idea!